test/functional/test-resources.js

```diff
@@ -1608,7 +1609,64 @@ describe('Resources changeSize', () => {
       expect(resources.relayoutTop_).to.equal(resource1.layoutBox_.top);
     });
 
-    it('should NOT adjust scrolling if size did not increase', () => {
+    it('should NOT resize when above vp but cannot adjust scrolling', () => {
```

Comment:
To compensate scrolling brought by height decrease, the best we can do is setting scrollTop to 0. Can't compensate here means scrolling jump can only be compensated by setting scrollTop to a negative value. 

In real use cases. Say a document with height 1000, the very top element has height 110, and viewport scrollTop is set to 100. According to the [rule](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1077) The top element is treated to be above viewport. However to avoid scrolling jump, we will need to set scrollTop to -10 in order to collapse this element. Which is not possible so resize should not happen.

In this specific test. viewport scrollTop is mocked to be 2, the resize element top position is -1200, and height is 100. In order to change its height to 0, we will need to set viewport scrollTop to -98 to compensate.